api/tests/integration/infrastructure/repositories/authentication-method-repository_test.js

```diff
@@ -606,40 +815,56 @@ describe('Integration | Repository | AuthenticationMethod', () => {
       await authenticationMethodRepository.removeByUserIdAndIdentityProvider({ userId, identityProvider });
 
       // then
-      const result = await knex('authentication-methods').where({ userId, identityProvider }).first();
+      const result = await knex('authentication-methods').where({ id: 123 }).first();
       expect(result).to.be.undefined;
     });
   });
 
-  describe('#findByUserId', () => {
+  describe('#findByUserId', function() {
 
-    it('should return the user\'s authentication methods', async () => {
+    it('should return the user\'s authentication methods', async function() {
       // given
       const userId = databaseBuilder.factory.buildUser().id;
-      databaseBuilder.factory.buildAuthenticationMethod({
+      const secondAuthenticationMethod = domainBuilder.buildAuthenticationMethod({
+        id: 456,
         identityProvider: AuthenticationMethod.identityProviders.GAR,
         externalIdentifier: 'externalIdentifier',
-        userId });
-      databaseBuilder.factory.buildAuthenticationMethod.buildWithHashedPassword({ userId });
+        userId,
+      });
+      secondAuthenticationMethod.authenticationComplement = undefined;
+      databaseBuilder.factory.buildAuthenticationMethod(secondAuthenticationMethod);
+      const firstAuthenticationMethod = domainBuilder.buildAuthenticationMethod.buildWithHashedPassword({
+        id: 123,
+        userId,
+        hashedPassword: 'Hello',
+      });
+      databaseBuilder.factory.buildAuthenticationMethod.buildWithHashedPassword({
+        ...firstAuthenticationMethod,
+        hashedPassword: 'Hello',
+      });
       await databaseBuilder.commit();
 
       // when
-      const result = await authenticationMethodRepository.findByUserId({ userId });
+      const authenticationMethods = await authenticationMethodRepository.findByUserId({ userId });
 
       // then
-      expect(result.length).to.equal(2);
+      expect(authenticationMethods).to.have.length(2);
+      expect(authenticationMethods[0]).to.be.instanceOf(AuthenticationMethod);
+      expect(authenticationMethods[1]).to.be.instanceOf(AuthenticationMethod);
+      expect(authenticationMethods[0]).to.deep.equal(firstAuthenticationMethod);
+      expect(authenticationMethods[1]).to.deep.equal(secondAuthenticationMethod);
```

Comment:
J'dis ça, j'dis rien
```suggestion
      expect(authenticationMethods[0]).to.deepEqualInstance(firstAuthenticationMethod);
      expect(authenticationMethods[1]).to.deepEqualInstance(secondAuthenticationMethod);
```

Reply:
Ah mince, c'est apparu après les helpers, je corrige !